pyproject.toml

```diff
@@ -28,7 +28,7 @@ classifiers = [
   "Programming Language :: Python :: Implementation :: PyPy",
   "Typing :: Typed",
 ]
-dependencies = []
+dependencies = ["license-expression"]
```

Comment:
PEP 517 backend have a huge restriction in terms of dependencies (no cycles) and most of them resort to vendoring `packaging`.

Therefore, if possible I would like to ask to please not add dependencies to `packaging`...

Probably somework based on the dowloads and local cache like proposed on https://github.com/pypa/packaging/pull/799 (with no dependency) would be ideal.

Reply:
This is related to the discussion in the Python discourse.
Relevant comments:

> [quote="Anderson Bravalheri, post:37, topic:53020, username:abravalheri"]
if it is possible for a backend to implement the validation/parsing (or any kind of processing needed) without getting in any extra dependency (or having to reimplementing it from scratch), then the issue with feasibility is gone, and then I have no problem with using a `SHOULD`.
[/quote]
> https://discuss.python.org/t/pep-639-round-3-improving-license-clarity-with-better-package-metadata/53020/37

> [quote="Pradyun Gedam, post:35, topic:53020, username:pradyunsg"]
I’ll say that, the current tentative implementation approach seems to be to end up implementing the parser for this in packaging, which does meet your “already vendored” mention in the footnote-rendered-inline thingie.
[/quote]
> https://discuss.python.org/t/pep-639-round-3-improving-license-clarity-with-better-package-metadata/53020/35

Reply:
I suspected that would be the case. Dependencies are tough in core tools like these.

The approach in #799 only loads the SPDX list, but doesn't implement a parser. https://github.com/aboutcode-org/license-expression and its only direct dependency https://github.com/bastikr/boolean.py are pure python and a great candidate for vendoring if the licensing is compatible.

Reply:
[986fe87](https://github.com/pypa/packaging/pull/828/commits/986fe87fde9f498e434602b5728f2e56bfd07fd6) setups up [`vendoring`](https://pypi.org/project/vendoring/) and [660fb2b](https://github.com/pypa/packaging/pull/828/commits/660fb2b61ae092b037133b41e421fac183918c94) actually vendors the dependencies.

license-expression is Apache-2.0, with a public domain vendored library of its own
boolean.py is BSD-Clause-2

I believe these are compatible but am not 100% certain.